### PR TITLE
Add support for chained panic handlers

### PR DESCRIPTION
**Pull request by PeterIvanov**
There is a test to illustrate an issue.
If error is not simply propagated along the same call chain by direct returns or panic, it is very useful to see all the places it changed hands. In some real life cases, error may cause panic and be handled as panic payload more than once, and having only the original stack trace is not enough. 
This PR adds more information here. 
As a negative note, more clutter in debug output will now be visited upon some simpler cases of panic handling. Panic is not a proper way to handle 'normal' errors in Go, so it is suggested to err on the side of greater verbosity here.

### CODE REVIEW

File: panic_test.go [hunk 1]
```diff
@@ -111,3 +111,68 @@ func funcWithBadPanic() {
...
+	err := <-ch
+	Panic(Decorate(err, "handle"))
+}
+
+func doMischief(ch chan error) {
+	defer func() {
+		if e := recover(); e != nil {
+			err, ok := ErrorFromPanic(e)
+			if ok {
+				ch <- Decorate(err, "hop 1")
```
Comment (g7r) [inline panic_test.go]: Why are you using a channel to return value from this function? Wouldn't it be easier to just return something? Yes, I see `defer` here. You will need named return apparently.

File: panic_test.go [hunk 2]
```diff
@@ -111,3 +111,68 @@ func funcWithBadPanic() {
...
+				return
+			}
+		}
+		ch <- AssertionFailed.New("test failed") // todo check
```
Comment (g7r) [inline panic_test.go]: How do you plan to address this todo? We need a clarification here.

File: panic_test.go (hunk 1 above)
Comment (PeterIvanov) [reply panic_test.go]: Fair point, it is a test half complete. The idea is to make "hops" as different in stack trace as possible - and so I want to spawn some goroutines. 
It all comes together via a simple write-read chain, and test itself has a timeout, so I don't see much of a danger here.

File: panic_test.go (hunk 2 above)
Comment (PeterIvanov) [reply panic_test.go]: I'm going to add an assertion against this type of error in return.

**Pull request merged**